Classes/common/CDTFetchChanges.m

```diff
@@ -46,19 +69,29 @@ - (void)main
         .sortBySequence = TRUE};
     
     TD_RevisionList *changes;
-    SequenceNumber lastSequence = [_startSequenceValue longLongValue];
+    SequenceNumber lastSequence = [self.mStartSequenceValue longLongValue];
 
     do {
-        changes = [[_datastore database] changesSinceSequence:lastSequence
-                                                      options:&options
-                                                       filter:nil
-                                                       params:nil];
+        if (self.cancelled) {
+            return;  // Bail early, don't call completion block
+        }
+
+        changes = [[self.mDatastore database] changesSinceSequence:lastSequence
+                                                           options:&options
+                                                            filter:nil
+                                                            params:nil];
         lastSequence = [self notifyChanges:changes startingSequence:lastSequence];
     } while (changes.count > 0);
 
-    void (^f)(NSString *nsv, NSString *ssv, NSError *fe) = self.fetchRecordChangesCompletionBlock;
+    void (^f)(NSString *nsv, NSString *ssv, NSError *fe) = self.mFetchRecordChangesCompletionBlock;
```

Comment:
I kept this one because I thought it make the if a bit clearer, but I think actually it doesn't now I come back to it.